src/jl_uv.c

```diff
@@ -681,11 +602,20 @@ int jl_printf(uv_stream_t *s, const char *format, ...)
     return c;
 }
 
-char *jl_bufptr(ios_t *s)
+DLLEXPORT void jl_safe_printf(const char *fmt, ...)
 {
-    return s->buf;
+    static char buf[1000];
+
+    va_list args;
+    va_start(args, fmt);
+    // Not async signal safe on some platforms?
+    vsnprintf(buf, sizeof(buf), fmt, args);
+    va_end(args);
+
+    write(STDERR_FILENO, buf, strlen(buf));
```

Comment:
linux man per link above says "The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str."
i.e. the '\0' is included is the written bytes.

POISX says "a null byte is written at the end of the bytes actually written into the array." -- http://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html#

MS do their own thing. #9624 is related.
Perhaps it's better to have a compliant wrapper for vsnprintf() on win32 rather than make every call site aware of all the MS weirdness ?